google-http-client-apache-v2/src/test/java/com/google/api/client/http/apache/v2/ApacheHttpTransportTest.java

```diff
@@ -175,4 +182,32 @@ public void process(HttpRequest request, HttpContext context)
     }
     assertTrue("Expected to have called our test interceptor", interceptorCalled.get());
   }
+
+  @Test
+  public void testNormalizedUrl() throws IOException {
+    int port = 8000 + new Random().nextInt(1000);
```

Comment:
Looks like I can use `0` for the port and it will pick a port https://docs.oracle.com/javase/7/docs/api/java/net/InetSocketAddress.html#InetSocketAddress(int)